spec/sequencer_spec.rb

```diff
@@ -337,10 +337,13 @@
         described_class.new(sequence_table, opts)
       end
 
+      before do
+        allow(sequence_table).to receive(:sequence_exists)
+        allow(cache).to receive(:get)
+      end
+
       it 'verify SequenceTable#truncate!' do
-        expect(sequence_table).to receive(:sequence_exists)
```

Comment:
Does this not get called anymore? trying to work out why this was moved

Reply:
its moved into the before block as an allow so the stub is setup before the initialise method is called, so it no longer makes real requests. `allow(sequence_table).to receive(:sequence_exists)`. I dont think this was noticed before because of the rescue standard error than logged nothing, until this pr

Reply:
Could we have both? as `expect` will fail if the method is not called, but `allow` wouldn't. I'm not sure why there is/was a test expecting that method to be called, but if it's part of the defined behaviour perhaps we should keep it in. There shouldn't be any clash in having both.